__init__.py

```diff
@@ -222,7 +271,13 @@ def patch(
         noise = torch.cat((ref_noise, noise), dim=0)
 
         x0_output = {}
-        callback = latent_preview.prepare_callback(m, sigmas.shape[-1] - 1, x0_output)
+        preview_callback = latent_preview.prepare_callback(m, sigmas.shape[-1] - 1, x0_output)
+
+        def callback(step: int, x0: T, x: T, steps: int):
+            preview_callback(step, x0, x, steps)
+            if (step < steps):
+                x[0] = ref_latents[step]
+                x0[0] = ref_latents[step]
```

Comment:
The reason I overwrite both the output and the reference sample is that this callback is apparently invoked by Comfy _before_ the sampling step subtracts the predicted output from the sample as the predicted noise, and uses that to make its next adjustment based on the particular sampler algorithm chosen; so we want the predicted noise to be 0, and avoid any _additional_ adjustment after we manually set the desired step output.